portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,30 +34,49 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private List<String> comments;
-  @Override
-  public void init() { 
-     comments = new ArrayList<>();
-  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    String json = new Gson().toJson(comments);
+    // initialize comments list and query
+    List<String> commentsList = new ArrayList<>();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    
+    // call datastore query to retrieve Comment Entity's
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    //iterate through Comment Entity's retrieved and add comments to list
+    for (Entity entity : results.asIterable()) {
+      String commentFromData = (String) entity.getProperty("comment");
+        System.out.println(commentFromData);
+      commentsList.add(commentFromData);
+    }
+
+    // convert comments list to JSON and send it to main page
```

Comment:
Great use of comments throughout this function. This'll make it way easier for people to work with your code (and for you to remember what you intended to do with your code when you revisit it weeks/months/years later!)

Reply:
great to hear! ill make this a habit